### PR TITLE
Support mTLS with The Things Join Server

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Follow up on https://github.com/TheThingsNetwork/lorawan-stack/pull/6031

#### Changes

This adds mTLS support to the TTJS claimer in DCS.

#### Testing

Unit testing only. Local integration testing TBD, but you can already review.

##### Regressions

None expected; configuration should be backwards compatible.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by KrishnaIyer — APPROVED** [review]
Changes look good to me

**Comment by KrishnaIyer** [comment]
One thing I remember is that the console E2E tests make some assumptions about the TTJS and add some stubs. I’m not sure if that’s still the case but that may need to be looked at.

File: pkg/deviceclaimingserver/enddevices/util_test.go [hunk 1]
```diff
@@ -16,89 +16,21 @@ package enddevices
...
-	"go.thethings.network/lorawan-stack/v3/pkg/httpclient"
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
-	"go.thethings.network/lorawan-stack/v3/pkg/types"
-	"go.thethings.network/lorawan-stack/v3/pkg/util/test"
-	"go.thethings.network/lorawan-stack/v3/pkg/web"
 	"google.golang.org/grpc"
 )
 
-var errMethodUnavailable = errors.DefineUnimplemented("method_unavailable", "method unavailable")
+// var errMethodUnavailable = errors.DefineUnimplemented("method_unavailable", "method unavailable")
```
Comment (adriansmares) [inline pkg/deviceclaimingserver/enddevices/util_test.go]: Did you mean to delete this too ?

**Review by adriansmares — APPROVED** [review]

**Review by nicholaspcr — APPROVED** [review]

File: pkg/deviceclaimingserver/enddevices/util_test.go (hunk 1 above)
Comment (johanstokking) [reply pkg/deviceclaimingserver/enddevices/util_test.go]: Yes, will do

**Comment by johanstokking** [comment]
@KrishnaIyer @adriansmares pushed some additional changes, please review the last three commits (https://github.com/TheThingsNetwork/lorawan-stack/pull/6039/files/38531b041207380819fc952fc5a8294ae5ac1dd3..719a2ef2afb61491a8ea355f1c5600c26eca6df0)

**Review by KrishnaIyer — APPROVED** [review]
The mentioned commits LGTM

**Pull request merged**